homeassistant/components/schluter/climate.py

```diff
@@ -0,0 +1,167 @@
+"""Support for Schluter thermostats."""
+import logging
+
+from requests import RequestException
+import voluptuous as vol
+
+from homeassistant.components.climate import (
+    PLATFORM_SCHEMA,
+    SCAN_INTERVAL,
+    ClimateDevice,
+)
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    HVAC_MODE_HEAT,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, CONF_SCAN_INTERVAL
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from . import DATA_SCHLUTER_API, DATA_SCHLUTER_SESSION, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {vol.Optional(CONF_SCAN_INTERVAL): vol.All(vol.Coerce(int), vol.Range(min=1))}
+)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the Schluter thermostats."""
```

Comment:
Make sure to return if discovery info is not None (someone accidentally added a platform config for it)

```suggestion
async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
    """Set up the Schluter thermostats."""
    if discovery_info is not None:
        return
```

Reply:
Merged this, needed a slight change to the code as I was passing in {} for the discovery parameter in load_platform. Thanks

Reply:
Yeah that is fine because `{}` is not None.

Reply:
Hi @balloob , my apologies if I am failing to understand. With your code suggestion, {} would return because it is not none. Is there a chance that the conditional should be 
`if discovery_info is None:` instead of `if discovery_info is not None:`

I was looking at a few other existing components and they seem to use
`if discovery_info is None:`

For instance, line 17 in this file
https://github.com/home-assistant/core/blob/206f8cef5c47c154354c0196b256b5feedf9478f/homeassistant/components/mychevy/binary_sensor.py